src/lib.rs

```diff
@@ -58,16 +59,45 @@ use iron::method::Method;
 use iron::status;
 use iron::headers;
 
+/// The struct that builds a CorsMiddleware
+pub struct CorsMiddlewareBuilder {
+    allowed_hosts: Option<HashSet<String>>,
+    allow_credentials: bool
+}
+
+impl CorsMiddlewareBuilder {
+    pub fn new() -> Self {
+        CorsMiddlewareBuilder { allow_credentials: false, allowed_hosts: None }
+    }
+
+    /// Specify which origin hosts are allowed to access the resource.
+    pub fn allowed_hosts(&mut self, allowed_hosts: HashSet<String>) -> &mut Self {
+        self.allowed_hosts = Some(allowed_hosts);
+        self
+    }
+    /// Specify Access-Control-Allow-Credentials
+    pub fn allow_credentials(&mut self, allow_credentials: bool) -> &mut Self {
+        self.allow_credentials = allow_credentials;
+        self
+    }
+
+    pub fn build(&self) -> CorsMiddleware {
+        CorsMiddleware { allowed_hosts: self.allowed_hosts.clone(), allow_credentials: self.allow_credentials }
+    }
```

Comment:
Can you change the `build` method to consume `self`? Then you can avoid the `clone()`.